### PR TITLE
Create URL with custom settings

### PR DESCRIPTION
**Pull request by irsheep**
Generates a URL with the user custom configuration settings, the URL can be used to re-open the page with the same configuration.

The settings are converted from JSON to a CSV and encoded to a Base64 string, I took this approach as encoding the JSON object was creating a large URL.
When encoding to base64 some characters are replaced so the encoded string is URL save.

Example:
https://beta.xkpasswd.net/?c=ZW5nbGlzaCw0LDQsOCxDQVBJVEFMSVNFLEZJWEVELExRLElVQWtKVjRtS2kxZkt6MDZmSDRfTHk0NywyLDIsTk9ORSw4LFJBTkRPTSwsSVVBa0pWNG1LaTFmS3owNmZINF9MeTQ3LDIsMg

### CODE REVIEW

File: src/web/settingsview.mjs [hunk 1]
```diff
@@ -13,13 +14,19 @@ class SettingsView {
    */
   #aniTime = 250;
 
+  /**
+   * @private {string} savedSettingsLink - set time to show/hide elements
```
Comment (hepabolu) [inline src/web/settingsview.mjs:18]: Please update the explanation of the savedSettingsLink. It now looks like a copy/paste error from the docs of aniTime

**Review by hepabolu — CHANGES_REQUESTED** [review]
At first, thanks a lot for this functionality. It looks very promising and a useful addition to the app.

I've tried your pull request and I found some issues. Please note that I have no intentions of turning this addition down, if my wording implies this, please ignore and regard it as an issue on my part.

The issues and suggestions below, are intended to improve the code in the pull request and make it more consistent with the current code base. If you don't want to spend more time, I'm more than happy to take it on.

CleanShot 2024-06-22-14-26@2x

- when I press either the icon or the word 'copy' and then try to paste it in a new browser window I get 'undefined' as link. This also happens when I select the link first and then press the 'copy button'
- I think I found a bug: I selected the 'temporary' preset and changed the words to 3, see image below
CleanShot 2024-06-22-14-44@2x

If I then copy/paste the url to a new window, I get a stack trace

CleanShot 2024-06-22-14-47@2x

Some suggestions:
- You added the 'custom' preset, which was also my idea to display when a custom preset was imported (either through the menu option or your link). So great idea 😉 However, it would be cool if you could also indicate that in the accordion bar, like the other presets. UPDATE: I can see that the could should already handle that, but it doesn't show up.

-  You created a separate config module to handle this link. Great stuff, but you put it in the lib directory. I may not have indicated this very well, but the intention is that the lib directory contains all the stuff that will be in the CLI library. 
Right now the config module is used in the index.mjs to handle the url during startup. In my mind this means the config module should be placed in the web directory.

**Comment by hepabolu** [comment]
Thanks for the feature

**Comment by hepabolu** [comment]
@irsheep This feature allows me to consolidate all loading and saving of configurations in one place. Thanks a lot! I need to rearrange and rename the code to match the other classes, but it will definitely keep its prominent functionality.

**Comment by podfeet** [comment]
@irsheep I know this issue is closed but I wanted to tell you that in today's episode of Programming By Stealth,  @hepabolu taught @bbusschots and me about how XKPasswd-js has implemented Model View Controller, you got a huge shout out during the episode. at about 1 hour and 3 minutes, she told us about your PR and what a fantastic solution it was to give us a way to save a configuration. Then she went on to explain how because of MVC, it made it easier and cleaner to implement into the bigger project. 

I wanted to thank you for your contribution and to make sure you heard the show and the nice things said about you. At the link for the show notes for Programming By Stealth 171, you can listen to the whole thing, or to jump to 1:03 to hear us talk about your work.

https://pbs.bartificer.net/pbs171

**Pull request merged**